pkg/sql/opt_exec_factory.go

```diff
@@ -2007,7 +2004,7 @@ func (ef *execFactory) ConstructAlterTableUnsplitAll(index cat.Index) (exec.Node
 func (ef *execFactory) ConstructAlterTableRelocate(
 	index cat.Index, input exec.Node, relocateSubject tree.RelocateSubject,
 ) (exec.Node, error) {
-	if !ef.planner.ExecCfg().IsSystemTenant() {
+	if relocateSubject != tree.RelocateLease && !ef.planner.ExecCfg().IsSystemTenant() {
```

Comment:
Check `relocateSubject != tree.RelocateLease` temporarily until #93987 removes this block.